tests/DependencyInjection/SonataNewsExtensionTest.php

```diff
@@ -0,0 +1,113 @@
+ */
+
+namespace Sonata\NewsBundle\Tests\DependencyInjection;
+
+use Matthias\SymfonyDependencyInjectionTest\PhpUnit\AbstractExtensionTestCase;
+use Sonata\EasyExtendsBundle\Mapper\DoctrineCollector;
+use Sonata\MediaBundle\Model\Media;
+use Sonata\NewsBundle\DependencyInjection\SonataNewsExtension;
+use Sonata\NewsBundle\Model\Comment;
+use Sonata\NewsBundle\Model\Post;
+use Sonata\UserBundle\Model\User;
+
+class SonataNewsExtensionTest extends AbstractExtensionTestCase
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->container->setParameter('kernel.bundles', []);
+    }
+
+    /**
+     * Test if the deprecation notice is trigger when the tag (or collection) class declaration is missing.
+     * It should trigger a deprecation notice but doesn't break anything
+     * You should have 0 associations.
+     *
+     * @group legacy
+     * @expectedDeprecation The class %s is not defined or doesn't exist. This is tolerated now but will be forbidden in 4.0
+     */
+    public function testLoadWithoutTagWithoutCollection(): void
+    {
+        $this->load($this->getMinimalConfiguration());
+        $collector = DoctrineCollector::getInstance();
+
+        $this->assertEmpty($collector->getAssociations());
+    }
+
+    /**
+     * Test if the deprecation notice is trigger when the tag (or collection) class declaration is present.
+     * It shouldn't trigger a deprecation notice but doesn't break anything
+     * You should have 2 associations (Post, Comment).
+     * The Post model should have an associations with (Media, User, Collection, Tag).
+     */
+    public function testLoadWithTagWithCollection(): void
+    {
+        $minimalConfiguration = $this->getMinimalConfiguration();
+        $tagAndCollectionDeclaration = [
+                'class' => [
+                        'collection' => \stdClass::class,
+                        'tag' => \stdClass::class,
+                ],
+            ];
+        $this->load(array_merge($minimalConfiguration, $tagAndCollectionDeclaration));
+        $collector = DoctrineCollector::getInstance();
+        //assert our model have associations
+        $this->assertCount(2, $collector->getAssociations());
+        $postManyToOneAssociation = $collector->getAssociations()[Post::class]['mapManyToOne'];
+        //assert the post model has  3 many to one associations (user,media,collection)
+        $this->assertCount(3, $postManyToOneAssociation);
+        $postManyToManyAssociation = $collector->getAssociations()[Post::class]['mapManyToMany'];
+        //assert the post model has  1 many to many associations (tag)
+        $this->assertCount(1, $postManyToManyAssociation);
+        $postOneToManyAssociation = $collector->getAssociations()[Post::class]['mapOneToMany'];
+        //assert the post model has  1 one to many associations (comment)
+        $this->assertCount(1, $postOneToManyAssociation);
```

Comment:
What do you think about transforming comments into $message arguments?
```suggestion
        $this->assertCount(
            1,
            $postOneToManyAssociation,
            'The post model should have 1 one to many association (comment)'
        );
```

Reply:
you are right, done !